src/components/NicknameForm.tsx

```diff
@@ -0,0 +1,99 @@
+import { InputAdornment, TextField, Typography, Button } from '@mui/material'
+import 'react-toastify/dist/ReactToastify.css'
+import { useForm, SubmitHandler } from 'react-hook-form'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { z } from 'zod'
+import { FaAngleRight } from 'react-icons/fa6'
+import { useState, useEffect } from 'react'
+import { toast } from 'react-toastify'
+import { SettingFormsProps } from '~/lib/types'
+
+const nicknameSchema = z.object({
+	nickname: z
+		.string()
+		.min(1, { message: 'Your nickname cannot be empty.' })
+		.regex(/^[a-zA-Z]/, { message: 'Start your name with a letter.' })
+		.min(5, { message: 'Make sure your name is at least 5 characters long.' })
+		.max(15, { message: 'Keep your name within 15 characters.' }),
+})
+
+interface NicknameFormInput {
+	nickname: string
+}
+
+const NicknameForm = ({ setIsDetailsVisible }: SettingFormsProps) => {
+	const [isFocusedField, setIsfocusedField] = useState(false)
```

Comment:
Out of curiosity, can you use `setFocus` from the useForm hook?
https://react-hook-form.com/docs/useform/setfocus

Reply:
I tried to use `setFocus`, but to unfocus I would have to use `ref` probably, and I still needed the focus state to dynamically render the `InputAdornment`